src/verify_cert.rs

```diff
@@ -431,20 +534,20 @@ const EKU_SERVER_AUTH: KeyPurposeId = KeyPurposeId::new(&oid!(1, 3, 6, 1, 5, 5,
 // id-kp-clientAuth   OBJECT IDENTIFIER ::= { id-kp 2 }
 const EKU_CLIENT_AUTH: KeyPurposeId = KeyPurposeId::new(&oid!(1, 3, 6, 1, 5, 5, 7, 3, 2));
 
-fn loop_while_non_fatal_error<V>(
+fn loop_while_non_fatal_error<'a, V: 'a>(
     default_error: Error,
     values: V,
-    mut f: impl FnMut(V::Item) -> Result<(), ControlFlow<Error, Error>>,
-) -> Result<(), ControlFlow<Error, Error>>
+    mut f: impl FnMut(V::Item) -> Result<&'a TrustAnchor<'a>, ControlFlow<Error, Error>>,
+) -> Result<&'a TrustAnchor<'a>, ControlFlow<Error, Error>>
 where
     V: IntoIterator,
 {
     let mut error = default_error;
     for v in values {
         match f(v) {
-            Ok(()) => return Ok(()),
+            Ok(item) => return Ok(item),
             // Fatal errors should halt further looping.
-            res @ Err(ControlFlow::Break(_)) => return res,
+            Err(ControlFlow::Break(err)) => return Err(ControlFlow::Break(err)),
```

Comment:
Was there a reason to drop the `res` binding? It seems less duplicative than reconstructing the return type from the `err` match (and IIRC it was a suggestion from Ctz in prev. reviews).